### PR TITLE
webpack unused + webpack dashboard

### PR DESCRIPTION
**Pull request by chrisnojima**
New webpack plugins to 

- [x] Show what files aren't being used
- [x] Show a nice dashboard of our webpack build hot server

@keybase/react-hackers

### CODE REVIEW

File: desktop/npm-helper.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ const commands = {
     help: 'Start electron with no hot reloading',
   },
   'build-dev': {
-    env: {NO_SERVER: 'true', DEBUG: 'express:*'},
```
Comment (chrisnojima) [inline desktop/npm-helper.js]: we don't need these express debug things anymore. webpack is stable and we don't need to see the individual requests

File: desktop/server.js [hunk 2]
```diff
@@ -21,10 +21,12 @@ if (NO_SERVER) {
     if (jsonStats.errors.length > 0) {
       throw jsonStats.errors.join('\n')
     }
+
+    console.log(stats)
+
     if (jsonStats.warnings.length > 0) {
```
Comment (chrisnojima) [inline desktop/server.js]: the unused files are a warning so lets just let that through and output it. don't die

File: desktop/webpack.config.development.js [hunk 3]
```diff
@@ -22,6 +25,31 @@ config.output.publicPath = 'http://localhost:4000/dist/'
 // Uncomment below to figure out packaging bugs
 // config.bail = true
 
+config.plugins.push(new UnusedFilesWebpackPlugin({
+  pattern: '../shared/**/*.js',
```
Comment (chrisnojima) [inline desktop/webpack.config.development.js]: have to do this else the paths won't match up due to symlink issues. its solvable by forking the plugin but it also makes it slower to resolve them

File: desktop/webpack.config.development.js [hunk 4]
```diff
@@ -22,6 +25,31 @@ config.output.publicPath = 'http://localhost:4000/dist/'
...
+  globOptions: {
+    ignore: [
+      // Mobile stuff
+      '../**/*.native.js',
+      '../**/*.ios.js',
+      '../**/*.android.js',
+      // Flow stuff
+      '../shared/constants/folders.js',
```
Comment (chrisnojima) [inline desktop/webpack.config.development.js]: these things are only used in a flow context so they're invisible to webpack so we just ignore them

File: shared/.flowconfig [hunk 5]
```diff
@@ -27,7 +27,7 @@
 ../desktop
 
 [libs]
-./libs/flow-interface.js
+./libs/flow-interface.js.flow
```
Comment (chrisnojima) [inline shared/.flowconfig]: lets make the flow-interface.js a flow file

File: shared/config/config.desktop.js [hunk 6]
```diff
@@ -1,11 +0,0 @@
-// @flow
```
Comment (chrisnojima) [inline shared/config/config.desktop.js]: 🔥

File: shared/dev/dumb-component-map.desktop.js [hunk 7]
```diff
@@ -1,5 +1,6 @@
 // @flow
 import CommonMap from '../common-adapters/dumb.desktop'
+import RegisterMap from '../login/register/dumb'
```
Comment (chrisnojima) [inline shared/dev/dumb-component-map.desktop.js]: was used in native but not desktop, whoops

File: shared/login/register/gpg-missing-pinentry/index.js [hunk 8]
```diff
@@ -1,15 +0,0 @@
-// @flow
-import React, {Component} from 'react'
```
Comment (chrisnojima) [inline shared/login/register/gpg-missing-pinentry/index.js]: 🔥

File: shared/login/signup/dumb.native.js [hunk 9]
```diff
@@ -1,6 +1,4 @@
 // @flow
-import Success from './success/dumb'
```
Comment (chrisnojima) [inline shared/login/signup/dumb.native.js]: this should just use the desktop one

File: shared/login/welcome/forgot-user-pass.native.js [hunk 10]
```diff
@@ -1,85 +0,0 @@
-// @flow
```
Comment (chrisnojima) [inline shared/login/welcome/forgot-user-pass.native.js]: these super old screens don't exist anymore

File: desktop/server.js (hunk 2 above)
Comment (cjb) [inline desktop/server.js]: Are there any unused files after your PR? I wonder if we should add a test to CI to avoid adding new ones.

Comment (chrisnojima) [inline desktop/server.js]: so this only shows up if you do a ```npm run build-dev``` so while devving we likely won't see it so CI is likely a good answer there. That seems fine to me if you guys are up for it. it means if that happens either

1. Real files got orphaned and should be deleted
1. or we added a file webpack doesn't know about (like flow-only files) and we need to add to the ignore list

if that seems ok to you guys lets ticket it

**Comment by awendland** [comment]
👍 Now to make a dashboard of every facet of Keybase

**Pull request merged**